includes/admin/admin-actions.php

```diff
@@ -709,7 +709,10 @@ function give_donation_import_callback() {
 	$import_setting['dry_run']     = $output['dry_run'];
 
 	// Parent key id.
-	$main_key = maybe_unserialize( $output['main_key'] );
+	$main_key = is_serialized( $output['main_key'] )
+		/** @unreleased Avoid insecure usage of `unserialize` when the data could be submitted by the user. */
+		? @unserialize( trim( $output['main_key'] ), ['allowed_classes' => false] )
+		: $output['main_key'];
```

Comment:
Looking at the suggestion and PHP [docs](https://www.php.net/manual/en/function.unserialize.php) I think it might be better to do json_encode for the input `name="main_key"` and do json_decode here

Reply:
Resolving this, for now.

> [...] using allowed_classes is a good enough patch for now (until more time for refactoring/testing can be found)